### PR TITLE
feat: micro experiment search placeholder

### PR DESCRIPTION
**Pull request by AmarTrebinjac**
## Changes
Introduced a feature for experimenting with search text, particularly placeholder in this case.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-564

### Preview domain
https://mi-564.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx [hunk 1]
```diff
@@ -68,14 +70,12 @@ export const SearchPanelInput = ({
     onFocus: externalOnFocus,
     onBlur: externalOnBlur,
     onClick: externalOnClick,
-    placeholder = searchPanel.isActive
-      ? 'Search posts or ask a question...'
-      : 'Search',
```
Comment (nensidosari) [inline packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:73]: Let's make sure here they want to experiment with both of these versions 🙏

Comment (AmarTrebinjac) [reply packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:73]: > Let's make sure here they want to experiment with both of these versions 🙏

You mean you want the experiment to only be displayed when the search panel is "active"? 😄

Comment (nensidosari) [reply packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:73]: Well, yeah that's what we need to check with product in this case, highlight to them that in the meantime we have 2 placeholders for when the panel is active and when panel is not active. 
Just because with the changes we're doing now in this PR, we're defaulting to only one placeholder for both.

Comment (nensidosari) [reply packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:73]: As for where to check this with product, u can either do on product-chatter channel, or in the experiment approval thread you added in experiments channel by tagging @product

File: packages/shared/src/lib/featureManagement.ts [hunk 2]
```diff
@@ -30,6 +30,9 @@ const feature = {
   onboardingChecklist: new Feature('onboarding_checklist', true),
   searchUsers: new Feature('search_users', false),
   showCodeSnippets: new Feature('show_code_snippets', false),
+  searchText: new Feature('search_text', {
+    placeholder: 'Search',
+  }),
```
Comment (sshanzel) [inline packages/shared/src/lib/featureManagement.ts:35]: Normally, we resort to using just a string when it comes to copy changes so this can be just a string literal.
```suggestion
  searchText: new Feature('search_text', 'Search'),
```

**Review by sshanzel — COMMENTED** [review]
Looks amazing! Just about how we normally define the values. We use object for complex experiments such as onboarding where we are trying to change multiple things at once.

File: packages/shared/src/lib/featureManagement.ts (hunk 2 above)
Comment (nensidosari) [reply packages/shared/src/lib/featureManagement.ts:35]: @sshanzel I created that just in case product wants to play with different parts of search text. But we can convert if we want to I don't mind.

Comment (nensidosari) [reply packages/shared/src/lib/featureManagement.ts:35]: Changed to string only since it doesn't seem product want to experiment differently 

https://app.growthbook.io/features/search_placeholder

flag is search_placeholder and is a string defaulting to Search

cc. @AmarTrebinjac

File: packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx [hunk 3]
```diff
@@ -44,7 +46,7 @@ export type SearchPanelInputClassName = {
 export type SearchPanelInputProps = {
   className?: SearchPanelInputClassName;
   valueChanged?: (value: string) => void;
-  inputProps?: InputHTMLAttributes<HTMLInputElement>;
+  inputProps?: Omit<InputHTMLAttributes<HTMLInputElement>, 'placeholder'>;
```
Comment (nensidosari) [inline packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:49]: why this change?

Comment (nensidosari) [reply packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:49]: Hmm, seems ok to remove actually since it's not being used

**Review by nensidosari — APPROVED** [review]
Looks good. Please before merging, update the experiment message with the new values of feature flag

File: packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx (hunk 3 above)
Comment (AmarTrebinjac) [reply packages/shared/src/components/search/SearchPanel/SearchPanelInput.tsx:49]: @nensidosari Like you said, it's not being used, so omitting it as to not have it show up as an available property, giving the impression that it does something  😅
